index.js

```diff
@@ -242,53 +240,78 @@ const parseArgs = (config = { __proto__: null }) => {
+      groupCount = expanded.length;
       continue;
     }
 
     if (isShortOptionAndValue(arg, options)) {
       // e.g. -fFILE
       const shortOption = StringPrototypeCharAt(arg, 1);
-      const longOption = findLongOptionForShort(shortOption, options);
-      const optionValue = StringPrototypeSlice(arg, 2);
-      checkOptionUsage(longOption, optionValue, options, `-${shortOption}`, strict, allowPositionals);
-      storeOption(longOption, optionValue, options, result.values);
+      const optionName = findLongOptionForShort(shortOption, options);
+      const value = StringPrototypeSlice(arg, 2);
+      ArrayPrototypePush(
+        elements,
+        { kind: 'option', optionName, optionUsed: `-${shortOption}`,
+          index, value, inlineValue: true });
       continue;
     }
 
     if (isLoneLongOption(arg)) {
       // e.g. '--foo'
-      const longOption = StringPrototypeSlice(arg, 2);
-      let optionValue;
-      if (optionsGetOwn(options, longOption, 'type') === 'string' &&
+      const optionName = StringPrototypeSlice(arg, 2);
+      let value;
+      let inlineValue;
+      if (optionsGetOwn(options, optionName, 'type') === 'string' &&
           isOptionValue(nextArg)) {
         // e.g. '--foo', 'bar'
-        optionValue = ArrayPrototypeShift(remainingArgs);
-        checkOptionLikeValue(longOption, optionValue, arg, strict);
+        value = ArrayPrototypeShift(remainingArgs);
+        inlineValue = false;
       }
-      checkOptionUsage(longOption, optionValue, options,
-                       arg, strict, allowPositionals);
-      storeOption(longOption, optionValue, options, result.values);
+      ArrayPrototypePush(
+        elements,
+        { kind: 'option', optionName, optionUsed: arg,
+          index, value, inlineValue });
       continue;
     }
 
     if (isLongOptionAndValue(arg)) {
       // e.g. --foo=bar
       const index = StringPrototypeIndexOf(arg, '=');
-      const longOption = StringPrototypeSlice(arg, 2, index);
-      const optionValue = StringPrototypeSlice(arg, index + 1);
-      checkOptionUsage(longOption, optionValue, options, `--${longOption}`, strict, allowPositionals);
-      storeOption(longOption, optionValue, options, result.values);
+      const optionName = StringPrototypeSlice(arg, 2, index);
+      const value = StringPrototypeSlice(arg, index + 1);
+      ArrayPrototypePush(
+        elements,
+        { kind: 'option', optionName, optionUsed: `--${optionName}`,
+          index, value, inlineValue: true });
       continue;
     }
 
-    // Anything left is a positional
-    if (!allowPositionals) {
-      throw new ERR_PARSE_ARGS_UNEXPECTED_POSITIONAL(arg);
-    }
-
-    ArrayPrototypePush(result.positionals, arg);
+    ArrayPrototypePush(elements, { kind: 'positional', index, value: arg });
   }
 
+  const result = {
+    values: { __proto__: null },
+    positionals: [],
+    // tokens: elements,
+  };
+  ArrayPrototypeForEach(elements, (element) => {
```

Comment:
Any reason not to use a `for of` here?

Reply:
I recall it was discouraged in a previous PR: https://github.com/pkgjs/parseargs/pull/63#discussion_r816212618